src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.java

```diff
@@ -366,17 +366,10 @@ else if (state.currentScopeState == STATE_STATIC_VARIABLE_DEF) {
                 state.currentScopeState = STATE_INSTANCE_VARIABLE_DEF;
             }
         }
-        else {
-            if (state.currentScopeState > STATE_STATIC_VARIABLE_DEF) {
-                if (!ignoreModifiers
-                        || state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF) {
-                    isStateValid = false;
-                    log(modifierAst, MSG_STATIC);
-                }
-            }
-            else {
-                state.currentScopeState = STATE_STATIC_VARIABLE_DEF;
-            }
+        else if (state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF
+                || state.currentScopeState > STATE_STATIC_VARIABLE_DEF && !ignoreModifiers) {
+            isStateValid = false;
+            log(modifierAst, MSG_STATIC);
```

Comment:
@Kevin222004 Shouldn't it be:
```java
else if (state.currentScopeState > STATE_STATIC_VARIABLE_DEF            
        && (!ignoreModifiers                                            
            || state.currentScopeState > STATE_INSTANCE_VARIABLE_DEF)) {
    isStateValid = false;                                               
    log(modifierAst, MSG_STATIC);                                       
}                                                                       
```

Reply:
Removal of `else` condition is justified but why have you changed the boolean operators.

Reply:
@Vyom-Yadav https://github.com/checkstyle/checkstyle/pull/13123#discussion_r1216506547 please gave a look at this conversation it is done for simplification purposes.

Reply:
Ok, https://github.com/checkstyle/checkstyle/pull/13123#discussion_r1214777249. 
Resolved.